Tests/otlLib/optimize_test.py

```diff
@@ -0,0 +1,172 @@
+import logging
+from pathlib import Path
+from subprocess import run
+import contextlib
+import os
+from typing import List, Optional, Tuple
+from fontTools.ttLib import TTFont
+
+import pytest
+
+from fontTools.feaLib.builder import addOpenTypeFeaturesFromString
+from fontTools.fontBuilder import FontBuilder
+
+from fontTools.ttLib.tables.otBase import OTTableWriter, ValueRecord
+
+
+def test_main(tmpdir: Path):
+    """Check that calling the main function on an input TTF works."""
+    glyphs = ".notdef space A Aacute B D".split()
+    features = """
+    @A = [A Aacute];
+    @B = [B D];
+    feature kern {
+        pos @A @B -50;
+    } kern;
+    """
+    fb = FontBuilder(1000)
+    fb.setupGlyphOrder(glyphs)
+    addOpenTypeFeaturesFromString(fb.font, features)
+    input = tmpdir / "in.ttf"
+    fb.save(str(input))
+    output = tmpdir / "out.ttf"
+    run(
+        [
+            "fonttools",
+            "otlLib.optimize",
+            "--gpos-compact-mode",
+            "5",
+            str(input),
+            "-o",
+            str(output),
+        ],
+        check=True,
+    )
+    assert output.exists()
+
+
+# Copy-pasted from https://stackoverflow.com/questions/2059482/python-temporarily-modify-the-current-processs-environment
+# TODO: remove when moving to the Config class
+@contextlib.contextmanager
+def set_env(**environ):
```

Comment:
I guess this function could be useful elsewhere, maybe move to misc.testTools?

Reply:
I think this is needed only for this PR now because it's passing around an environment variable, but as soon as there's a Config class in fontTools, the environment variable should go and the associated test too.

Maybe we should clarify somewhere that the environment variable is not a stable feature and that it will be removed eventually? So that if projects start using this feature in their pipelines, they know that at some point they will have to turn it on via the Config class instead?

Or is there the argument that if we merge this with the ENV variable, then it will need to work with the ENV variable forever, even once the Config class is introduced?

Reply:
yeah you can mark it as experimental if you like. we can remove support once it becomes a burden

Reply:
How do I mark as experimental?

Reply:
just add a comment along those lines (and we can also mention that in the NEWS.rst when we update that)